cmd/juju/application/refresher/refresher.go

```diff
@@ -162,14 +162,18 @@ func (d *localCharmRefresher) Allowed(_ RefresherConfig) (bool, error) {
 	return true, nil
 }
 
+var charmNameChangeError = `
+By refreshing %q to %q it looks like you're changing the charm itself, which is not supported.
```

Comment:
```suggestion
Refreshing %q to %q is not supported, use --force flag to override.
```

Reply:
I'm not sure about this, the problem is we don't _know_ if what the user is doing is supported or not. They could be trying to refresh to a compatible charm, or they may not

Reply:
What you had wasn't ideal, speak to @tmihoc

Reply:
Looking at https://canonical-juju.readthedocs-hosted.com/en/latest/user/howto/manage-charms/#update-a-local-charm we see that `juju refresh --path` is related to `juju refresh --switch`, and the latter is the recommended way to go if you are trying to replace the charm with a completely different charm.

So here I would say:

> Judging by the name of the charm that you provided, it looks like you're cross-grading to a different charm. If that is so, please use --switch.

Speaking of which, I'm also finding the help for `juju refresh --force` confusing:

```
--force  (= false)
    Allow a charm to be refreshed which bypasses LXD profile allow list
```

Why the specific mention of LXD profiles?